src/engine/SCons/compat/__init__.py

```diff
@@ -98,7 +98,7 @@ def rename_module(new, old):
 
 # Was pickle.HIGHEST_PROTOCOL
 # Changed to 2 so py3.5+'s pickle will be compatible with py2.7.
-PICKLE_PROTOCOL = 2
+PICKLE_PROTOCOL = -2
```

Comment:
How did you get -2? Wouldn't pickle.HIGHEST_PROTOCOL be an alternate fix?

Reply:
a few lines above, someone mentioned that negative number would pick the highest available:
https://github.com/SCons/scons/blob/ad792d38e8f3cabf7da8aa990d1c29c6b79bb41f/src/engine/SCons/compat/__init__.py#L94-L97

I thought this would be more compatible across python versions. I did not test pickle.HIGHEST_PROTOCOL.

Reply:
So the reason 2 was specified is that it's compatible with both 2.7 and 3.x.  By choosing any negative number that effectively becomes HIGHEST_PROTOCOL, but in a much less readable way.  Can you change to HIGHEST_PROTOCOL?   Still curious why this fixes the issue.